### PR TITLE
Feature/testing

### PR DESCRIPTION
**Pull request by Javier-M-V**

### CODE REVIEW

**Comment by wen96** [comment]
The branch has conflicts with dev. Can you merge dev into it or rebase it with dev?

File: api/tests/services/test_steam_api_service.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        # Arrange
+        mock_api_key.return_value = 'hummus'
+        steam_id = 1234
+        url = "http://api.steampowered.com/ISteamUserStats/GetUserStatsForGame/v0002/"
+        url_params = "?appid=730&key=hummus&steamid=1234"
+        super_url = "{}{}".format(url, url_params)
+        #  Act
+        service = SteamAPIService()
+        service.get_cs_info(steam_id)
+        mock_request_endpoint.assert_called_with(super_url)
```
Comment (Javier-M-V) [inline api/tests/services/test_steam_api_service.py]: Solved

File: api/tests/services/test_steam_api_service.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        result = service.get_steam_id_from_nick_name(nickname)
+
+        # Asssert
+        self.assertIsNone(result)
+
+    @mock.patch.object(SteamAPIService, '_request_endpoint')
+    @mock.patch.object(SteamAPIService, 'api_key')
+    def test_get_steam_id_from_nick_name_returns_steam_id(self, mock_api_key, mock_request_endpoint):
+        # Arrange
+        mock_api_key.return_value = 'hummus'
```
Comment (Javier-M-V) [inline api/tests/services/test_steam_api_service.py]: Solved

File: api/tests/services/test_steam_api_service.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        # Act
+        service = SteamAPIService()
+        result = service.get_steam_id_from_nick_name(nickname)
+
+        # Asssert
+        self.assertIsNone(result)
+
+    @mock.patch.object(SteamAPIService, '_request_endpoint')
+    @mock.patch.object(SteamAPIService, 'api_key')
```
Comment (Javier-M-V) [inline api/tests/services/test_steam_api_service.py]: Solved

File: api/tests/services/test_steam_api_service.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+class SteamApiServiceTestCase(TestCase):
+
+    @mock.patch.object(SteamAPIService, '_request_endpoint')
+    @mock.patch.object(SteamAPIService, 'api_key')
+    def test_get_steam_id_from_nick_name_returns_none_if_user_not_found_in_reponse(
+            self, mock_api_key, mock_request_endpoint):
+        # Arrange
+        mock_api_key.return_value = 'hummus'
```
Comment (Javier-M-V) [inline api/tests/services/test_steam_api_service.py]: Solved

File: api/tests/services/test_steam_api_service.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import mock
+
+from django.test import TestCase
+from api.services import SteamAPIService
+
+
+class SteamApiServiceTestCase(TestCase):
+
+    @mock.patch.object(SteamAPIService, '_request_endpoint')
+    @mock.patch.object(SteamAPIService, 'api_key')
```
Comment (Javier-M-V) [inline api/tests/services/test_steam_api_service.py]: Solved

**Pull request merged**